ledger-core/v2/scripts/gen_plantuml.go

```diff
@@ -0,0 +1,562 @@
+    } else {
+
+        for _, fld := range fn.Recv.List {
+
+            // Receiver
+            recv := &RecvPair{
+                Name: fld.Names[0].Name,
+                Type: fmt.Sprintf("%s", pf.code[fld.Type.Pos()-1:fld.Type.End()-1]),
+            }
+
+            // Parameters
+            pars := make(map[string]string, 0)
+            for _, par := range fn.Type.Params.List {
+                if nil == par.Names {
+                    pars["unnamed-param"] = fmt.Sprintf("%s", pf.code[par.Type.Pos()-1:par.Type.End()-1])
+                } else {
+                    pars[par.Names[0].Name] = fmt.Sprintf("%s", pf.code[par.Type.Pos()-1:par.Type.End()-1])
+                }
+            }
+
+            // I want to analyse only methods, who takes context
+            if !isMethodTakesCtx(pars) {
+                pf.diag("\n:parseMethod: skip %s - Doesn`t take CTX", fn.Name.Name)
+                continue
+            }
+
+            // I want analyse only methods, which returned values
+            if nil == fn.Type.Results {
+                pf.diag("\n:parseMethod: skip %s - No return value", fn.Name.Name)
+                continue
+            }
+
+            // I want to analyze methods which have a `smashine.StateUpdate' result type
+            res := fn.Type.Results.List[0].Type
+            resSel, ok := res.(*ast.SelectorExpr)
+            if !ok || "StateUpdate" != resSel.Sel.Name {
+                if pf.dbg {
+                    fmt.Printf("\n:parseMethod: skip %s - No StateUpdate result type", fn.Name.Name)
+                }
+                continue
+            }
+            resXstr := fmt.Sprintf("%s", pf.code[resSel.X.Pos()-1:resSel.X.End()-1])
+            if "smachine" != resXstr {
+                if pf.dbg {
+                    fmt.Printf("\n:parseMethod: skip %s - No smachine selector result type", fn.Name.Name)
+                }
+                continue
+            }
+
+            // Show name (debug)
+            pf.diag("\n:parseMethod: (sm-name) %s", fn.Name.Name)
+
+            // Find all Return Statements and SetDefaultMigration calls
+            var rets = make([]*Ret, 0)
+            var migr = ""
+            for _, smth := range fn.Body.List { // ∀ fn.Body.List ← (or RetStmt (Inspect ...))
+                retStmt, ok := smth.(*ast.ReturnStmt)
+                if ok {
+                    // return from top-level statements of function
+                    rets = append(rets, pf.collectRets(retStmt, "Top")...)
+                } else {
+                    ast.Inspect(smth, func(in ast.Node) bool {
+                        // Find Return Statements
+                        retStmt, ok := in.(*ast.ReturnStmt) // ←
+                        if ok {
+                            // return from deep-level function statememt
+                            rets = append(rets, pf.collectRets(retStmt, "Deep")...)
+                        } else {
```

Comment:
this could be 
```
if ok {
return true
}
...
```
without else, less indentation, much easier to read

and best practice in go is ot write 
```
if _, ok := something(); !ok {
    return
}
...
```
keeping low cyclomatic complexity